client/views/admin/users/UserForm.js

```diff
@@ -68,6 +68,10 @@ export default function UserForm({
 		[],
 	);
 
+	const roleKeys = Array.from(new Map(availableRoles).keys());
+	const rolesToShow = roles.filter((r) => roleKeys.indexOf(r) >= 0);
+	useEffect(() => handleRoles(rolesToShow), []);
```

Comment:
We should try to avoid this `useEffect` it's missing the dependencies!

Reply:
Thanks for the review @dougfabris . I am on it. :)

For the 1st point, I have made a separate pr, please have a look at #20955 . I didn't see it was reviewed by gazzo until now, I will fix that pr asap and also add the empty value check before showing custom fields. 

As for the JSON validation check I am not quite sure how we can do it from front-end. I think we can do it at backend, when we try to save the json. For the front end, we can add a new type of `settings` which would take only json, but I will have to think more about this Idea.

What do you think, should I do changes of pr #20955 here or let it be there as its already under review by sir gazzo?

Thanks :)

Reply:
@yash-rajpal Let's keep it separated PR, please check Gazzo's review on #20955 
About JSON validation, I will find a way to check the JSON code when the admin saves the configuration as well